src/server/scripts/Northrend/Naxxramas/instance_naxxramas.cpp

```diff
@@ -212,6 +218,22 @@ class instance_naxxramas : public InstanceMapScript
                 case NPC_LICH_KING:
                     _lichkingGUID = creature->GetGUID();
                     return;
+                case NPC_PATCHWORK_GOLEM:
+                    PatchwerkRoomTrash.push_back(creature->GetGUID());
+                    return;
+                case NPC_BILE_RETCHER:
+                    if (creature->GetPositionY() > -3258.0f) // we want only those inside the room, not before
```

Comment:
This is a bit hacky, uses a magic number.

Reply:
can't be declared otherwise, this is the Y coord of the entrance of the 'room' patchwerk is in, else we will pull all the trash in both rooms (before and where patchwerk is)